crates/ra_hir/src/name.rs

```diff
@@ -72,6 +72,14 @@ impl Name {
         };
         Some(name)
     }
+
+    pub fn as_str(&self) -> &str {
+        self.text.as_str()
+    }
+
+    pub(crate) fn to_string(&self) -> String {
+        self.text.to_string()
+    }
```

Comment:
We shouldn't convert `Name`s to string unless it's for display purposes, and, in that case, there's an `Display` purpose. Let's remove these couple methods.

Inside HIR, `Name`s can be compared directly (and that should account for hygiene, once we have that).